File: ML/mmax.m
function [outputArg1] = mmax(p)
p = p*sign(p(1));
a=0;
flag=1;
while (flag == 1)
  sum=0;
  a = a + 1;
  flag = 0;
  if (p(1) <= 0)
   flag = 1;
  end
  for i=1:length(p)
    sum = sum*a;
    sum = sum + p(i);
    if (sum < 0)
      flag = 1;
    end
  end
end
outputArg1 = a;
end

File: Python/mmax.py
def mmax(p):
    a = 0
    flag = 1
    while (flag == 1):
        sum = 0
        a += 1
        flag = 0
        if p[0] <= 0:
            flag = 1
        for i in range(0, len(p)):
            sum *= a
            sum += p[i]
            if sum < 0:
                flag = 1
    return a

if __name__ == "__main__":
    p = [-7, 4, 8]
    print(my_top(p))


        
